mmdet/models/losses/huber_loss.py

```diff
@@ -0,0 +1,46 @@
+# Based off of github.com/rwightman/efficientdet-pytorch
+
+import torch
+import torch.nn as nn
+
+from ..builder import LOSSES
+from .utils import weighted_loss
+
+@weighted_loss
+def huber_loss(pred, target, delta):
```

Comment:
What is the difference between huber_loss and smooth_l1_loss?

Reply:
huber_loss returns 0.5 * quadratic.pow(2) + delta * linear, where quadratic is either abs_err or delta if abs_err > delta, and linear is abs_err - quadratic, and delta is an inputted value

smooth_l1_loss returns 0.5 * diff * diff / beta if diff < beta, or diff - 0.5 * beta, where beta is an inputted value

Between these two cases, diff and abs_err are both the absolute value of pred - target

Reply:
Let's try to see the forest for the trees. Could you please use the same naming for the `abs(pred - target)` and simplify expressions for huber loss. I believe it'll make it clearer what the actual difference is.

Reply:
Updated huber_loss, let me know if there should be more simplification

Reply:
No need to make any changes to the code for now.
You still not convinced me that these two losses are any different in a mathematical sense. Difference in multiplicative constant does not count, because it can be easily compensated via proper `loss_weight` value.

Reply:
Found this thread about smooth l1 loss vs huber loss: https://github.com/torch/nn/issues/579

Reply:
Great. And what is the conclusion?

Reply:
That thread decided that you can't replicate huber loss using smooth l1. While there are similarities (top comment mentioned that smooth l1 acts like huber with delta=1), there are enough differences (such as different value of delta) that it would be worthwhile to add

Reply:
Cool. The next question is, how does that thread relate to this PR?

Here is the task for you: Get mathematical formulas that both smooth_l1 and huber loss as they are implement here and prove that those differ more than just by a constant multiplier.

Reply:
This relates to this thread because HuberLoss is a new loss functions that was used in the Pytorch implementation of mmdetection that a lot of this work is based off of ([link](https://github.com/rwightman/efficientdet-pytorch/blob/master/effdet/loss.py#L87), and is currently being used by the added model config

The equations for huber loss and smooth l1 loss are (x is the absolute value of the difference between pred and target):

Huber loss
`0.5 * x^2                       when x <= d
0.5 * d^2 + d * (x - d)    when x > d
or d(x - 0.5 * d)              when x > d'

Where d is delta

Smooth l1 loss
'0.5 * x^2 / b                  when x < b
x - 0.5 * b                       when x >= b'

Where b is beta

Given there isn't much of a difference with the used version of huber loss (minus the multiplicative constants), will remove from PR and replace with smooth l1